### ISSUE TITLE
vfs: TestLock fails on Macs when run with a -race flag.

### ISSUE THREAD

**Issue by bananabrick** [cockroachdb/pebble#1504]
Command to reproduce: `go test -race -run TestLock`, under the `vfs` directory. Test was run on the master branch.

```
arjunnair@crlMBP-C02FR0UTMD6TMjA4 vfs % go test -race -run TestLock
vfs.test(11386,0x8b09600) malloc: *** error for object 0x600001700080: pointer being freed was not allocated
vfs.test(11386,0x8b09600) malloc: *** set a breakpoint in malloc_error_break to debug
SIGABRT: abort
PC=0x7ff802bc8112 m=0 sigcode=0

goroutine 0 [idle]:
syscall.rawSyscall(0x10c8180, 0x0, 0x0, 0x0)
        /usr/local/opt/go/libexec/src/runtime/sys_darwin.go:77 +0x47 fp=0xc0000c9570 sp=0xc0000c9550 pc=0x10a8d47
syscall.rawSyscall(0x10aee05, 0x0, 0x0, 0x1)
        <autogenerated>:1 +0x26 fp=0xc0000c95b8 sp=0xc0000c9570 pc=0x10af726
syscall.forkAndExecInChild(0xc0000a0660, {0xc000156150, 0x6, 0x1}, {0xc000128100, 0x20, 0x0}, 0x0, 0x0, 0xc0000c98c0, ...)
        /usr/local/opt/go/libexec/src/syscall/exec_libc2.go:80 +0xed fp=0xc0000c9648 sp=0xc0000c95b8 pc=0x10bd00d
syscall.forkExec({0x7ff7bfeff820, 0x51}, {0xc00009e780, 0x5, 0x5}, 0xc0000668c0)
        /usr/local/opt/go/libexec/src/syscall/exec_unix.go:212 +0x5a7 fp=0xc0000c97e0 sp=0xc0000c9648 pc=0x10be0a7
syscall.StartProcess(...)
        /usr/local/opt/go/libexec/src/syscall/exec_unix.go:264
os.startProcess({0x7ff7bfeff820, 0x51}, {0xc00009e780, 0x5, 0x5}, 0xc0000c9b50)
        /usr/local/opt/go/libexec/src/os/exec_posix.go:55 +0x526 fp=0xc0000c9918 sp=0xc0000c97e0 pc=0x10fdea6
os.StartProcess({0x7ff7bfeff820, 0x51}, {0xc00009e780, 0x5, 0x5}, 0x104818d)
        /usr/local/opt/go/libexec/src/os/exec.go:109 +0x7a fp=0xc0000c9978 sp=0xc0000c9918 pc=0x10fd33a
os/exec.(*Cmd).Start(0xc0000f49a0)
        /usr/local/opt/go/libexec/src/os/exec/exec.go:422 +0x9d4 fp=0xc0000c9ba8 sp=0xc0000c9978 pc=0x1344654
os/exec.(*Cmd).Run(0xc000066ca8)
        /usr/local/opt/go/libexec/src/os/exec/exec.go:338 +0x31 fp=0xc0000c9be8 sp=0xc0000c9ba8 pc=0x1343c11
os/exec.(*Cmd).CombinedOutput(0xc0000f49a0)
        /usr/local/opt/go/libexec/src/os/exec/exec.go:567 +0x20a fp=0xc0000c9c58 sp=0xc0000c9be8 pc=0x134556a
github.com/cockroachdb/pebble/vfs_test.spawn({0x7ff7bfeff820, 0x51}, {0xc0000a0420, 0x55})
        /Users/arjunnair/go/src/github.com/cockroachdb/pebble/vfs/file_lock_test.go:23 +0xfb fp=0xc0000c9cf8 sp=0xc0000c9c58 pc=0x146b39b
github.com/cockroachdb/pebble/vfs_test.TestLock(0xc00013e9c0)
        /Users/arjunnair/go/src/github.com/cockroachdb/pebble/vfs/file_lock_test.go:60 +0x4cb fp=0xc0000c9e98 sp=0xc0000c9cf8 pc=0x146b8eb
testing.tRunner(0xc00013e9c0, 0x150f770)
        /usr/local/opt/go/libexec/src/testing/testing.go:1259 +0x230 fp=0xc0000c9fb0 sp=0xc0000c9e98 pc=0x1188650
testing.(*T).Run·dwrap·21()
        /usr/local/opt/go/libexec/src/testing/testing.go:1306 +0x48 fp=0xc0000c9fe0 sp=0xc0000c9fb0 pc=0x1189be8
runtime.goexit()
        /usr/local/opt/go/libexec/src/runtime/asm_amd64.s:1581 +0x1 fp=0xc0000c9fe8 sp=0xc0000c9fe0 pc=0x10ac1a1
created by testing.(*T).Run
        /usr/local/opt/go/libexec/src/testing/testing.go:1306 +0x727

goroutine 19 [running]:
runtime.systemstack_switch()
        /usr/local/opt/go/libexec/src/runtime/asm_amd64.s:350 fp=0xc0000c9520 sp=0xc0000c9518 pc=0x10a9f60
runtime.libcCall(0x18, 0x17dd380)
        /usr/local/opt/go/libexec/src/runtime/sys_libc.go:49 +0x66 fp=0xc0000c9550 sp=0xc0000c9520 pc=0x1098266
syscall.rawSyscall(0x10c8180, 0x0, 0x0, 0x0)
        /usr/local/opt/go/libexec/src/runtime/sys_darwin.go:77 +0x47 fp=0xc0000c9570 sp=0xc0000c9550 pc=0x10a8d47
syscall.rawSyscall(0x10aee05, 0x0, 0x0, 0x1)
        <autogenerated>:1 +0x26 fp=0xc0000c95b8 sp=0xc0000c9570 pc=0x10af726
syscall.forkAndExecInChild(0xc0000a0660, {0xc000156150, 0x6, 0x1}, {0xc000128100, 0x20, 0x0}, 0x0, 0x0, 0xc0000c98c0, ...)
        /usr/local/opt/go/libexec/src/syscall/exec_libc2.go:80 +0xed fp=0xc0000c9648 sp=0xc0000c95b8 pc=0x10bd00d
syscall.forkExec({0x7ff7bfeff820, 0x51}, {0xc00009e780, 0x5, 0x5}, 0xc0000668c0)
        /usr/local/opt/go/libexec/src/syscall/exec_unix.go:212 +0x5a7 fp=0xc0000c97e0 sp=0xc0000c9648 pc=0x10be0a7
syscall.StartProcess(...)
        /usr/local/opt/go/libexec/src/syscall/exec_unix.go:264
os.startProcess({0x7ff7bfeff820, 0x51}, {0xc00009e780, 0x5, 0x5}, 0xc0000c9b50)
        /usr/local/opt/go/libexec/src/os/exec_posix.go:55 +0x526 fp=0xc0000c9918 sp=0xc0000c97e0 pc=0x10fdea6
os.StartProcess({0x7ff7bfeff820, 0x51}, {0xc00009e780, 0x5, 0x5}, 0x104818d)
        /usr/local/opt/go/libexec/src/os/exec.go:109 +0x7a fp=0xc0000c9978 sp=0xc0000c9918 pc=0x10fd33a
os/exec.(*Cmd).Start(0xc0000f49a0)
        /usr/local/opt/go/libexec/src/os/exec/exec.go:422 +0x9d4 fp=0xc0000c9ba8 sp=0xc0000c9978 pc=0x1344654
os/exec.(*Cmd).Run(0xc000066ca8)
        /usr/local/opt/go/libexec/src/os/exec/exec.go:338 +0x31 fp=0xc0000c9be8 sp=0xc0000c9ba8 pc=0x1343c11
os/exec.(*Cmd).CombinedOutput(0xc0000f49a0)
        /usr/local/opt/go/libexec/src/os/exec/exec.go:567 +0x20a fp=0xc0000c9c58 sp=0xc0000c9be8 pc=0x134556a
github.com/cockroachdb/pebble/vfs_test.spawn({0x7ff7bfeff820, 0x51}, {0xc0000a0420, 0x55})
        /Users/arjunnair/go/src/github.com/cockroachdb/pebble/vfs/file_lock_test.go:23 +0xfb fp=0xc0000c9cf8 sp=0xc0000c9c58 pc=0x146b39b
github.com/cockroachdb/pebble/vfs_test.TestLock(0xc00013e9c0)
        /Users/arjunnair/go/src/github.com/cockroachdb/pebble/vfs/file_lock_test.go:60 +0x4cb fp=0xc0000c9e98 sp=0xc0000c9cf8 pc=0x146b8eb
testing.tRunner(0xc00013e9c0, 0x150f770)
        /usr/local/opt/go/libexec/src/testing/testing.go:1259 +0x230 fp=0xc0000c9fb0 sp=0xc0000c9e98 pc=0x1188650
testing.(*T).Run·dwrap·21()
        /usr/local/opt/go/libexec/src/testing/testing.go:1306 +0x48 fp=0xc0000c9fe0 sp=0xc0000c9fb0 pc=0x1189be8
runtime.goexit()
        /usr/local/opt/go/libexec/src/runtime/asm_amd64.s:1581 +0x1 fp=0xc0000c9fe8 sp=0xc0000c9fe0 pc=0x10ac1a1
created by testing.(*T).Run
        /usr/local/opt/go/libexec/src/testing/testing.go:1306 +0x727

goroutine 1 [chan receive]:
testing.(*T).Run(0xc00013e820, {0x14f5a6e, 0x8}, 0x150f770)
        /usr/local/opt/go/libexec/src/testing/testing.go:1307 +0x752
testing.runTests.func1(0x0)
        /usr/local/opt/go/libexec/src/testing/testing.go:1598 +0x9a
testing.tRunner(0xc00013e820, 0xc00011fbf8)
        /usr/local/opt/go/libexec/src/testing/testing.go:1259 +0x230
testing.runTests(0xc000110400, {0x17d6640, 0x12, 0x12}, {0xc0001523f0, 0xc00011fc80, 0x17dce60})
        /usr/local/opt/go/libexec/src/testing/testing.go:1596 +0x7cb
testing.(*M).Run(0xc000110400)
        /usr/local/opt/go/libexec/src/testing/testing.go:1504 +0x9d2
main.main()
        _testmain.go:83 +0x22c

rax    0x0
rbx    0x8b09600
rcx    0x7ff7bfeff328
rdx    0x0
rdi    0x303
rsi    0x6
rbp    0x7ff7bfeff350
rsp    0x7ff7bfeff328
r8     0x0
r9     0x0
r10    0x0
r11    0x246
r12    0x303
r13    0x50
r14    0x6
r15    0x16
rip    0x7ff802bc8112
rflags 0x246
cs     0x7
fs     0x0
gs     0x0
--- FAIL: TestLock (0.01s)
    file_lock_test.go:52: Locking: /var/folders/_0/qnnnjwl94pgb6mcn71rlz3nw0000gp/T/golang-pebble-db-testlock-1763144577
    file_lock_test.go:59: Spawning child, should fail to grab lock.
    file_lock_test.go:65: Child failed with unexpected output: 
FAIL
exit status 1
FAIL    github.com/cockroachdb/pebble/vfs       0.287s
arjunnair@crlMBP-C02FR0UTMD6TMjA4 vfs % 
```

**Comment by nicktrav** [comment]
I think this is specific to macOS 12. Looks like it's fixed in the latest release of Go (1.17.7).

Fix, if you're interested:  https://github.com/golang/go/commit/5f6552018d1ec920c3ca3d459691528f48363c3c

**Comment by nicktrav** [comment]
Going to close this, as it was fixed upstream. The fix, if you hit this, is to update your Go version.
